lib/puppet/module/task.rb

```diff
@@ -61,6 +62,66 @@ def self.is_tasks_filename?(path)
       return true
     end
 
+    def self.get_file_details(path, name_start)
+      unless File.absolute_path(path) == File.path(path)
+        msg = _("File pathnames cannot include relative paths")
+        raise InvalidMetadata.new(msg, 'puppet.tasks/invalid-metadata')
+      end
+
+      # This gets the path from the starting point onward
+      # For files this should be the file subpath from the metadata
+      # For directories it should be the directory subpath plus whatever we globbed
+      # Partition matches on the first instance it finds of the parameter
+      name = path.partition(name_start).drop(1).join("")
+
+      { "name" => name, "path" =>  path }
+    end
+
+    # Find task's required lib files and retrieve paths
+    # for both 'files' and 'implementation:files' metadata keys
+    def self.find_files(files, mod)
+      env = mod.environment.respond_to?(:name) ? mod.environment.name : 'production'
+
+      file_list = files.flat_map do |file|
+        module_name, mount, endpath = file.split("/", 3)
+        # If there's a directory with no trailing slash, this will be nil
```

Comment:
I don't think this comment is true. `endpath` will preserve the trailing slash for something like `module/lib/dir/` because it'll produce `['module', 'lib', 'dir/']`, but it would only be nil if the path were incorrect like `module/lib`.

Reply:
Yes, that's what the comment says.

Reply:
But that's not true for `module/lib/dir`, because it only splits into 3 elements.

Reply:
Is `module/lib` and incorrect path? Do there need to be subdirs?

Reply:
I think that would be invalid. `lib` is a mount point and must be a directory.